docs/sources/shared/wal-data-retention.md

```diff
@@ -74,5 +74,32 @@ If the component falls behind more than one third of the data written since the
 last truncate interval, it is possible for the truncate loop to checkpoint data
 before being pushed to the remote_write endpoint.
 
-[WAL block]: {{< relref "../flow/reference/components/prometheus.remote_write.md/#wal-block" >}}
-[metrics config]: {{< relref "../static/configuration/metrics-config.md" >}}
+### WAL corruption
+
+WAL corruption can occur when Grafana Agent unexpectedly stops while the latest WAL segments
+are still being written to disk. For example, the host computer has a general disk failure
+and crashes before you can stop Grafana Agent and other running services. When you restart Grafana
+Agent, the Agent verifies the WAL, removing any corrupt segments it finds. Sometimes, this repair
+is unsuccessful, and you must manually delete the corrupted WAL to continue.
+
+When the WAL becomes corrupted, Grafana Agent writes error messages such as
+`err="failed to find segment for index"` to the log file.
+
+{{% admonition type="caution" %}}
+Deleting a WAL segment or a WAL file permanently deletes the stored WAL data.
+{{% /admonition %}}
```

Comment:
Stored WAL data isn't replayed on restart anyway, so this caution isn't that severe (at least until this behavior is changed)

Reply:
@rfratto I could convert it to a note... or remove it altogether.